coltypes/coltypes.go

```diff
@@ -17,117 +17,94 @@ package coltypes
 
 import (
 	"context"
-	"database/sql"
 	"encoding/json"
 	"fmt"
-)
-
-const (
-	jsonType = "json"
-
-	// boolAliasType is our custom alias to the tinyint(1).
-	boolAliasType = "bool"
-	boolType      = "tinyint(1)"
-
-	// Types that can be represented as Go strings.
-	charType       = "char"
-	textType       = "text"
-	longTextType   = "longtext"
-	mediumTextType = "mediumtext"
-	tinyTextType   = "tinytext"
-	varcharType    = "varchar"
-	timeType       = "time" // format is 15:04:34
-	enumType       = "enum"
-	setType        = "set"
-	decimalType    = "decimal"
-)
 
-var (
-	// querySchemaColumnTypes is a query that selects column names and
-	// their data and column types from the information_schema.
-	querySchemaColumnTypes = "select column_name, data_type, column_type " +
-		"from information_schema.columns where table_name = ?;"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/proto/query"
 )
 
-// Querier is a database querier interface needed for the GetColumnTypes function.
-type Querier interface {
-	QueryContext(ctx context.Context, query string, args ...any) (*sql.Rows, error)
-}
+// TransformRow converts the values of type sqltypes.Value to appropriate Go types, based on the fields parameter.
+// This is necessary because the underlying raw values are just byte slices.
+func TransformRow(ctx context.Context, fields []*query.Field, values []sqltypes.Value) (map[string]any, error) {
+	if len(fields) != len(values) {
+		return nil, ErrFieldsValuesLenMissmatch
+	}
 
-// TransformRow converts row map values to appropriate Go types, based on the columnTypes.
-// This is necessary because Vitess driver doesn't scan some types into a map[string]any correctly.
-// Some types become byte slices instead of strings or maps (JSON), etc.
-func TransformRow(ctx context.Context, row map[string]any, columnTypes map[string]string) (map[string]any, error) {
-	result := make(map[string]any, len(row))
+	result := make(map[string]any, len(fields))
 
-	for key, value := range row {
-		if value == nil {
-			result[key] = value
+	for i, field := range fields {
+		// seperate check for null type, since the field.Type may not match the values[i].Type()
+		if field.Type == query.Type_NULL_TYPE || values[i].Type() == query.Type_NULL_TYPE {
+			result[field.Name] = nil
 
 			continue
 		}
 
-		switch columnTypes[key] {
-		case jsonType:
-			valueBytes, ok := value.([]byte)
-			if !ok {
-				return nil, convertValueToBytesErr(key)
+		switch field.Type {
+		case query.Type_INT8, query.Type_INT16, query.Type_INT24,
+			query.Type_INT32, query.Type_INT64, query.Type_YEAR, query.Type_BIT:
+
+			// if the column length is 1,
+			// and the column type is the integral type of INT8 - the column is boolean
```

Comment:
Yes, it can be just a binary integer (`0` or `1`), but if a user creates a column with type `INT8`, which is the internal representation of the `TINYINT(1)`, he should understand that it's an alias to `BOOL`.

According to the MySQL docs: "_`BOOL` and `BOOLEAN` are synonyms of `TINYINT(1)`_".